deps/oob-1.0.2/jsonutil.c

```diff
@@ -404,7 +410,35 @@ void setoobConnectioninfo(const char* host,
     copyString("https", connectionInfo.protocol, 8);
   else
     copyString("http", connectionInfo.protocol, 8);
+}
 
+void setoobDsninfo(const char *key, const char *val) {
+    char* caps = calloc(strlen(key), sizeof(char));
+    upper(caps, key);
+
+    if (dsn == NULL) {
+        dsn = cJSON_CreateObject();
+    }
+    cJSON *value = cJSON_CreateString(val);
+    if (strcmp(caps, "PWD")) {
+        if (!strcmp(caps, "SERVER")) {
+            setdeployment(val);
+        }
+        cJSON_AddItemToObject(dsn, key, value);
+    } else {
+        cJSON_AddItemToObject(dsn, key, cJSON_CreateString("***"));
```

Comment:
I know you try to mask sensitive value. Question: do you limit the keys from DSN? If yes, where is the list? If no, that may be a security issue, e.g., if customer set `proxy_pwd` in the key value pair, we are going to miss it.